File: 1032.js
var input = require('fs').readFileSync('input.txt', 'utf8');
var lines = input.split('\n');

function prime(x){
    if(x === 1) return 0;
    if(x % 2 === 0) return (x === 2) ? 1 : 0;   
    for(let i=3; i<=Math.sqrt(x); i++){
        if(x % i === 0) return 0;
    }
    return 1;
}

function prime_list(n){
    let primes = [], i = 2;
    while(n > 0){
        if(prime(i)){
            primes.push(i);
            n--;
        }
        i++;
    }
    return primes;
}

var n = parseInt(lines.shift());
while(n > 0){
    let people = [];
    let primes = prime_list(n-1);
    
    for(let i=1; i<=n; i++) people.push(i);
    
    let curr = 0, k = 0;
    while(people.length > 1){
        curr = (curr + primes[k++] - 1) % people.length;
        people.splice(curr,1);
    }
    
    console.log(people[0]);
    var n = parseInt(lines.shift());
}

File: 1032.rb
def prime(x)
    if x == 1
        return 0
    end
    if x % 2 == 0
        return (x == 2) ? 1 : 0
    end
    for i in 3..Math.sqrt(x) do
        if x % i == 0
            return 0
        end
    end
    return 1
end

def prime_list(n)
    primes = []
    i = 2
    while n > 0
        if prime(i) == 1
            primes.push(i)
            n -= 1
        end
        i += 1
    end
    return primes
end

while (n = gets.chomp.to_i) > 0
    people = []
    for i in 1..n do
        people.push(i)
    end
    curr = 0
    k = 0
    primes = prime_list(n-1)
    while people.length > 1
        curr = (curr + primes[k].to_i - 1) % people.length
        people.delete_at(curr)
        k += 1
    end
    puts people[0]
end